src/pages/index.js

```diff
@@ -0,0 +1,627 @@
+  avatarInfo.querySelector(".avatar__description"),
+];
+
+const profileInfoValue = [
+  modalEditProfile.querySelector("#name"),
+  modalEditProfile.querySelector("#description"),
+];
+
+const modalNewPostValue = [
+  modalNewPost.querySelector("#picture"),
+  modalNewPost.querySelector("#caption"),
+];
+
+const modalAvatarImage = profile.querySelector(".avatar__image");
+
+const modalEditProfileAvatarImageInput =
+  modalEditProfileAvatar.querySelector("#picture");
+
+// // Event listeners to toggle edit profile and newpost modals on and off
+editProfileButton.addEventListener("click", () => {
+  resetErrorDisplayedInEditProfileModal(modalEditProfile, settings);
+  disableSubmitButton(submitProfileButton);
+  openModal(modalEditProfile);
+
+  // TODO: Get the values of each form field from the value property
+  // of the corresponding input element.
+  for (let i = 0; i < profileInfoValue.length; i++) {
+    profileInfoValue[i].value = profileInfoSave[i].textContent;
+  }
+});
+
+editProfileOverlayedButton.addEventListener("click", () => {
+  resetErrorDisplayedInEditProfileModal(modalEditProfileAvatar, settings);
+  disableSubmitButton(formEditProfileAvatar);
+  openModal(modalEditProfileAvatar);
+
+  // TODO: Get the values of each form field from the value property
+  // of the corresponding input element.
+});
+// Handle submitted profile function
+function handleEditProfileAvatarFormSubmit(evt) {
+  // Prevent default browser behavior, see explanation below.
+  evt.preventDefault();
+
+  // notify the user that the upload process is underway
+  // by changing the button text to "Saving..."
+  editProfileAvatarSubmitButton.textContent = "Saving...";
+
+  // TODO: Then insert these new values into the src property of the
+  // profile image.
+  modalAvatarImage.src = modalEditProfileAvatarImageInput.value;
+
+  // Send the following PATCH request to change the profile picture
+  api
+    .changeProfilePicture(modalEditProfileAvatarImageInput.value)
+    .then((result) => {
+      // process the result
+      console.log(result);
+    })
+    .catch((err) => {
+      console.error(err); // log the error to the console
+    })
+    .finally(() => {
+      // Change button text back to original
+      editProfileAvatarSubmitButton.textContent = "Save";
+
+      disableSubmitButton(editProfileAvatarSubmitButton);
+      formEditProfileAvatar.reset();
+      // TODO: Close the modal.
+      closeModal(modalEditProfileAvatar);
+    });
+}
+// Handle submitted profile function
+function handleProfileFormSubmit(evt) {
+  // Prevent default browser behavior, see explanation below.
+  evt.preventDefault();
+
+  // Change button text to "Saving..."
+  submitProfileButton.textContent = "Saving...";
+
+  // TODO: Then insert these new values into the textContent property of the
+  // corresponding profile elements.
+  for (let i = 0; i < profileInfoValue.length; i++) {
+    profileInfoSave[i].textContent = profileInfoValue[i].value;
```

Comment:
* You need to update the `DOM` only in blocks  `.then` after a successful response from the server, so that when there is an error, it shouldn’t change